icepyx/core/query.py

```diff
@@ -125,6 +125,8 @@ def __init__(
 
         if (
             dataset is None or spatial_extent is None or date_range is None
+        ) and (
+            cycles is None or tracks is None
```

Comment:
To submit a query, the user has to submit a dataset. Then, they can submit cycles and tracks OR date_range and spatial. Do we want to require them to submit spatial along with cycles and tracks (since these are otherwise pretty big)? I think that's the way to go, but am not set on it. Then this would be revised along the lines of `if (dataset is None or spatial_extent is None) and ((cycles is None or tracks is None) or (date_range is None)):`. I think the `files` part isn't yet implemented and that's what @andypbarrett was hoping to add, right?

Reply:
It does work without a spatial parameter, BUT any users would have to make sure `subset` is set to False when they create their order (since there's nothing to subset).  Might be better to just require a spatial parameter :+1:

Reply:
They could still subset with variables though... Sounds like we agree on requiring the spatial parameter, since otherwise you can end up with a lot more data than you bargained for (e.g. if you take the spatial parameter out of your example). It's easy enough to specify the whole globe if you really want the entire track.

Reply:
right I forget about the variable subsetting features.  Yep I think that adding back the spatial requirement sounds good :+1: